### DISCUSSION TITLE
Adding Kafka as a logging backend

### DISCUSSION THREAD

**Discussion by saryani** [jmpsec/osctrl#499] [Ideas]
Hello @javuto 🙋‍♂️

I’m with [Maersk](https://www.maersk.com/), and we’re interested in integrating osctrl into our product. However, we would like to have a feature that allows logging information into Kafka.

We’re willing to develop this feature and contribute it to osctrl, but we’d like to know if this is something you would find useful and would be willing to merge.

Please let us know, and we can proceed with the implementation

**Comment by javuto** [comment]
Hello @saryani 👋 

That sounds great! I am more than happy to support you with the implementation. You can find some examples on how to implement a new logger for `osctrl-tls` in https://github.com/jmpsec/osctrl/tree/main/logging 
My only question is do you also want to retrieve the logs from Kafka and display them in `osctrl-admin`? That would add complexity but we'll figure it out 😃 

Thank you!

**Reply by saryani** [reply]
Thanks @javuto 🙌

That's amazing 🎉  
Regarding your question, I think we do not need that feature, all we need is to be able to push agents logs into a Kafka topic.

**Reply by martinjaegerdk** [reply]
I agree - that's not a feature we need.

**Reply by saryani** [reply]
the PR for the above change is here: https://github.com/jmpsec/osctrl/pull/527
